react/src/components/CreateMapModal/CreateMapModal.tsx

```diff
@@ -0,0 +1,211 @@
+import React, { useState } from 'react';
+import {
+  Modal,
+  ModalHeader,
+  ModalBody,
+  ModalFooter,
+  FormGroup,
+  Label,
+  Input,
+} from 'reactstrap';
+import { Button } from '../../core-components';
+import styles from './CreateMapModal.module.css';
+import { Formik, Form, Field, ErrorMessage } from 'formik';
+import * as Yup from 'yup';
+import useCreateProject from '../../hooks/projects/useCreateProject';
+import useAuthenticatedUser from '../../hooks/user/useAuthenticatedUser';
+import { useNavigate } from 'react-router-dom';
+import { ProjectRequest } from '../../types';
+
+type CreateMapModalProps = {
+  isOpen: boolean;
+  toggle: () => void;
+};
+
+// Yup validation schema
+const validationSchema = Yup.object({
+  name: Yup.string().required('Name is required'),
+  description: Yup.string().required('Description is required'),
+  system_file: Yup.string()
+    .matches(
+      /^[A-Za-z0-9-_]+$/,
+      'Only letters, numbers, hyphens, and underscores are allowed'
+    )
+    .required(' file name is required'),
+});
+
+const CreateMapModal = ({ isOpen, toggle }: CreateMapModalProps) => {
+  const [errorMessage, setErrorMessage] = useState('');
+  const { data: userData } = useAuthenticatedUser();
+  const { mutate: createProject, isLoading: isCreatingProject } =
+    useCreateProject();
+  const navigate = useNavigate();
+
+  const handleCreateProject = (projectData: ProjectRequest, actions) => {
+    createProject(projectData, {
+      onSuccess: (newProject) => {
+        navigate(`/project/${newProject.uuid}`);
+      },
+      onError: (err) => {
+        // Handle error messages while creating new project
+        setErrorMessage('An error occurred while creating the project.');
+        if (err?.response?.status === 409) {
+          setErrorMessage(
+            'That folder is already syncing with a different map.'
+          );
+        } else if (err?.response?.status === 500) {
+          setErrorMessage('Internal server error. Please contact support.');
+        }
```

Comment:
what about just having two types of error messages? one for the 409 and one for just the general error?
```suggestion
        if (err?.response?.status === 409) {
          setErrorMessage(
            'That folder is already syncing with a different map.'
          );
        } else {
          setErrorMessage('An error occurred while creating the project. Please contact support.');
        }
```